design/open/acl/rbac-to-acl.md

```diff
@@ -0,0 +1,202 @@
+
+##### Grants
+
+A group is _granted_ a single permission within a scope.  "Admin" permission
+has no scope (grayed out in the GUI, and all repos are allowed).
+
+#### Out: RBAC
+
+Remove from the GUI the policy page and all references to policies, and from
+`lakectl` all policy-related commands.  The API will no longer support these
+operations, either.  It will return [405 Method Not Allowed][http-stat-405],
+or alternatively [501 Unimplemented][http-stat-501] when called: it is now a
+partial implementation of the auth* API.
+
+#### In: ACLs
+
+Every group on the list of groups on the Groups page uses the format:
+
+| Group ID | Permission          | Created at | Repositories |
+|----------|---------------------|------------|--------------|
+| <name>   | Permission dropdown | datetime   | <number>     |
+
+![Groups page has a dropdown to edit permission for each
+group](./groups-with-perms.png)
+
+This gives information about the group and allows editing its permission.
+
+There are 4 default groups, named after the 4 permissions.  Each group is
+global (applies for all repositories).
+
+Clicking on the group takes us to the group's subpage with 2 tabs:
+
+* **Members**: The current "Group Memberships" tab.
+* **Repositories**: A list of configured repositories.
+
+  This tab has an "all" toggle at the top.  If the permission is "Admin" the
+  toggle is set and cannot be changed.  Otherwise it controls a selection of
+  groups using the same GUI mechanism as used today for memberships (and for
+  policies, ironically).
+
+We shall add appropriate options to `lakectl` to control policies at the new
+levels of granularity.
+
+### Upgrade to simplified
+
+On startup, lakeFS will check the KV schema version.  If it has not upgraded
+to ACLs then lakeFS will check whether an upgrade is required:
+
+* If the 4 basic policies have not been modified, and no other policies have
+  been defined, and no users have attached policies: no upgrade is required.
+  lakeFS will merely upgrade the KV schema version.
+* Otherwise lakeFS will not start until an upgrade script runs.
+
+#### The upgrade script
+
+Admins will run the upgrade script if required to do so by lakeFS or if they
+want to be very sure (for instance, there is a minor race condition if there
+is more than one concurrent lakeFS).  We force admins to try the script on a
+dry run using a `--yes` flag: without it, the script only prints the planned
+changes.
+
+The upgrade script will _ensure_ that the 4 default global groups exist, and
+_modify_ existing groups to fit into the new ACLs scheme.
+
+* When creating the 4 default global groups: if another group exists and has
+  the desired name, upgrading will rename it by appending ".orig".  So after
+  upgrading the 4 default global groups exist, with these known names.
+* For any group, upgrading configured policies follows these rules, possibly
+  **increasing** access:
+
+  1. Any "Deny" rules are stripped, and a warning printed.
```

Comment:
please make sure the warning message contains a link to our slack workspace or support email if they need any assistance.